### PR TITLE
Add support for log rotate on Windows  & Version bump to 1.0.2

### PR DESCRIPTION
**Pull request by ridiculousness**
## Description
- [Windows|Linux] Updated installation to 1.0.2 
  Windows requires package_version because of the x86/x64 msi's.
  Note also that in the url there is a - character and in the package this - is a .
   
Windows:
  default["sensu"]["msi_repo_url"] = "http://repositories.sensuapp.org/msi/2016"
  default["sensu"]["version"] = "1.0.2-1-x64"
  default['sensu']['windows_package_version'] = '1.0.2.1'

Linux:
  default["sensu"]["version"] = "1.0.2-1"

- [Windows] Updated client xml to support log rotating the in the sensu log folder.
  ["sensu"]["log_directory"] Log folder can be any folder as logn as the file name is sensu-client.out

## Motivation and Context
Log file on Windows was not being rotated. This caused disk to fill up.
https://github.com/sensu/sensu-chef/issues/580

## How Has This Been Tested?
Tested this on Windows 2016  & CentOS 7, Chef client 12.19.36
Run Chef-client, see if it installs the newer version, check if the client.xml is being changed.
See if the log file is being rotated (tested with 10kb else I would have to wait for a much longer time)

## Screenshots (if appropriate):
afbeelding
Please note that the rotate was set to 10kb but after running Chef it was changed to 10 MB, therefor the sensu-client.out is larger than the rotated files.

## Types of changes

- [X] New feature (non-breaking change which adds functionality)

## Checklist:

- [X] My change requires a change to the documentation.
- [X] I have updated the documentation accordingly.
- [X] I have read the **CONTRIBUTING** document, but I do not have Kitchen etc installed.

### CODE REVIEW

**Comment by eheydrick** [comment]
The log changes look reasonable to me. If we're updating versions I'd suggest going to the latest release, 1.1.1. @cwjohnston @amdprophet @majormoses can you take a look too.

File: attributes/default.rb [hunk 1]
```diff
@@ -10,30 +10,33 @@
...
+default["sensu"]["log_rotate_file_keep"] = '10'
 default["sensu"]["use_ssl"] = true
 default["sensu"]["use_embedded_ruby"] = true
 default["sensu"]["service_max_wait"] = 10
 default["sensu"]["directory_mode"] = "0750"
 default["sensu"]["log_directory_mode"] = "0750"
 default["sensu"]["client_deregister_on_stop"] = false
 default["sensu"]["client_deregister_handler"] = nil
-
+default["sensu"]["msi_repo_url"] = "http://repositories.sensuapp.org/msi/2016"
```
Comment (majormoses) [inline attributes/default.rb]: This is a breaking change and needs to be called out in the changelog. Again I would prefer we limit the changes strictly to what is required to use logrotate.

File: attributes/default.rb [hunk 2]
```diff
@@ -10,30 +10,33 @@
   default["sensu"]["windows"]["package_options"] = nil
   default["sensu"]["windows"]["install_dotnet"] = true
   default["sensu"]["windows"]["dotnet_major_version"] = 4
+  default["sensu"]["version"] = "1.0.2-1-x64"
+  default['sensu']['windows_package_version'] = '1.0.2.1'
 else
   default["sensu"]["admin_user"] = "root"
   default["sensu"]["directory"] = "/etc/sensu"
   default["sensu"]["log_directory"] = "/var/log/sensu"
+  default["sensu"]["version"] = "1.0.2-1"
```
Comment (majormoses) [inline attributes/default.rb]: Is this required to use the new logrotate? I'd prefer to keep the changes as isolated as possible. I am not sure that we should default to 1.x yet as there are a handful of community plugins that have conflicts and need to be worked out first. You can always overwrite the version in your wrapper cookbook.

File: attributes/default.rb [hunk 3]
```diff
@@ -10,30 +10,33 @@
   default["sensu"]["windows"]["package_options"] = nil
   default["sensu"]["windows"]["install_dotnet"] = true
   default["sensu"]["windows"]["dotnet_major_version"] = 4
+  default["sensu"]["version"] = "1.0.2-1-x64"
```
Comment (majormoses) [inline attributes/default.rb]: Is this required to use the new logrotate? I'd prefer to keep the changes as isolated as possible. I am not sure that we should default to 1.x yet as there are a handful of community plugins that have conflicts and need to be worked out first. You can always overwrite the version in your wrapper cookbook.

File: attributes/default.rb [hunk 4]
```diff
@@ -10,30 +10,33 @@
...
+  default['sensu']['windows_package_version'] = '1.0.2.1'
```
Comment (majormoses) [inline attributes/default.rb]: Is this required to use the new logrotate? I'd prefer to keep the changes as isolated as possible. I am not sure that we should default to 1.x yet as there are a handful of community plugins that have conflicts and need to be worked out first. You can always overwrite the version in your wrapper cookbook.

File: attributes/default.rb (hunk 3 above)
Comment (ridiculousness) [reply attributes/default.rb]: Logrotate is supported as of 0.29.5 (https://github.com/sensu/sensu-chef/issues/580). I checked the repo url and it has been changed. If I look at http://repositories.sensuapp.org/msi/. I see 1 type of msi for both 32/64 bit, now there are 2 msi's (http://repositories.sensuapp.org/msi/2016/). I had to test this with 1.0.2 as we have had issues with pre 1.x versio and already installed 1.0.2, which was latest at that moment of time. Reverting back to 0.28.4-1 would mean that logrotate won't work.

File: attributes/default.rb (hunk 4 above)
Comment (ridiculousness) [reply attributes/default.rb]: This one is tricky.

Installing "1.0.2-1-x64" results in '1.0.2.1' in the registry. 

afbeelding

As it does not has the 1.0.2-1-x64 display version it installs the client every chef run. This should imo be fixed in the MSI itself to keep the MSI version the same as the display version in the Windows registry. If that happens the attribute is absolete.

File: attributes/default.rb (hunk 2 above)
Comment (ridiculousness) [reply attributes/default.rb]: Linux installation version does not come with a 32/64 bit version.

Previous it was all the same for both Windows and Linux.

File: attributes/default.rb (hunk 1 above)
Comment (ridiculousness) [reply attributes/default.rb]: The url wihtout the 2016 has no MSI above 0.26.5 :( so I had to change it. I expect but have not tested it that the msi's in the 2016 folder are either backwards compatible or the same as in the 2012R2 folder.

**Comment by ridiculousness** [comment]
@majormoses  Awaiting your comments based on my comments. If I need to change the version to 0.29.5 (the minimum release to support logrotate) I will do so.

File: attributes/default.rb (hunk 4 above)
Comment (majormoses) [reply attributes/default.rb]: ping @cwjohnston whos the best person to discuss the msi fix for version numbers?

**Comment by majormoses** [comment]
uggh this is messy. I am ok with these changes mostly. I'd like to hear from @cwjohnston first on some of the comments before merging and releasing.

**Comment by majormoses** [comment]
ping @cwjohnston

File: attributes/default.rb (hunk 3 above)
Comment (cwjohnston) [reply attributes/default.rb]: > I see 1 type of msi for both 32/64 bit

I believe these are 32bit builds which work on 64bit systems. We now build both 32 and 64 bit installers.

File: attributes/default.rb (hunk 4 above)
Comment (cwjohnston) [reply attributes/default.rb]: This `-x64` suffix may be an artifact introduced by the toolchain changes starting in Sensu 0.27. Opening an issue against https://github.com/sensu/sensu-omnibus would help us track and fix this.

File: attributes/default.rb (hunk 1 above)
Comment (cwjohnston) [reply attributes/default.rb]: On the repository server the 2016 and 2012r2 directories are symlinks to the 2012 directory. There's no difference in the content being served.

**Comment by majormoses** [comment]
closing per https://github.com/sensu/sensu-chef/pull/607#issue-181825797

**Pull request closed without merging**